api.proto

```diff
@@ -288,6 +298,7 @@ message Ack {
     AckPolicy ackPolicy          = 7; // The AckPolicy sent on the message
     int64     receptionTimestamp = 8 [jstype=JS_STRING]; // When the message was received by the server
     int64     commitTimestamp    = 9 [jstype=JS_STRING]; // When the message was committed
+    AckError  ackError           = 10; // AckError sent in case the partition refused to accept the published message
```

Comment:
I think the enum definition isn't right actually. Since an enum isn't nullable, this will always default to an `UNKNOWN` error if not set. See how it's done here: https://github.com/liftbridge-io/liftbridge-api/blob/1511cbd0018153167088518326ce45fa3af87d51/api.proto#L226-L229

Also, I think we should nest the enum in the `Ack` message similar to `StreamMetadata`.

Reply:
In that case, I proposed:

- 0 to be `OK`
- 1 to be anything unknown
- 2, 3, 4...etc to be specific errors.